onnxruntime/core/providers/cpu/sequence/sequence_ops.cc

```diff
@@ -209,10 +209,10 @@ Status SequenceInsert::Compute(OpKernelContext* context) const {
 
   const auto* I = context->Input<Tensor>(2);
   int64_t num_tensors_input_seq = static_cast<int64_t>(S->Size());
-  int64_t input_seq_idx = num_tensors_input_seq + 1;  // default is append
-  if (I) {                                            // position is optional
+  int64_t input_seq_idx = num_tensors_input_seq;  // default is append
+  if (I) {                                        // position is optional
     input_seq_idx = GetSeqIdx(*I);
-    if (!ValidateSeqIdx(input_seq_idx, static_cast<int64_t>(num_tensors_input_seq))) {
+    if (!ValidateSeqIdx(input_seq_idx, num_tensors_input_seq) && input_seq_idx != num_tensors_input_seq) {
```

Comment:
The check ```input_seq_idx != num_tensors_input_seq``` can go inside ValidateSeqIdx.

Reply:
This check is actually making an exception for SequenceInsert, which differs in allowing `input_seq_idx == num_tensors_input_seq`. Other way is to increase `num_tensors_input_seq` by 1 before calling `ValidateSeqIdx`, but that won't work with its logic of handling negative `input_seq_idx`. Thus additional check.